### PR TITLE
Clean Post Form Extraction

### PR DESCRIPTION
**Pull request by postables**
## :construction_worker: Purpose
clean up post form extraction and validation

## :rocket: Changes
Helper function that takes in slice of post forms needed, returns a mapping of `name -> value`

## :warning: Breaking Changes
None

### CODE REVIEW

File: api/routes_payment.go [hunk 1]
```diff
@@ -23,17 +23,8 @@ import (
...
-		FailWithMissingField(c, "payment_number")
-		return
-	}
-	txHash, exists := c.GetPostForm("tx_hash")
-	if !exists {
-		FailWithMissingField(c, "tx_hash")
-		return
-	}
-	paymentNumberInt, err := strconv.ParseInt(paymentNumber, 10, 64)
+	forms := api.extractPostForms([]string{"payment_number", "tx_hash"}, c)
```
Comment (postables) [inline api/routes_payment.go]: need to check length lenth

File: api/routes_payment.go [hunk 2]
```diff
@@ -71,13 +62,12 @@ func (api *API) ConfirmPayment(c *gin.Context) {
...
 func (api *API) RequestSignedPaymentMessage(c *gin.Context) {
 	username := GetAuthenticatedUserFromContext(c)
-	paymentType, exists := c.GetPostForm("payment_type")
-	if !exists {
-		FailWithMissingField(c, "payment_type")
-		return
-	}
-	var method uint64
-	switch paymentType {
+	forms := api.extractPostForms([]string{"payment_type", "sender_address", "credit_value"}, c)
```
Comment (postables) [inline api/routes_payment.go]: need to check length

File: api/utils.go [hunk 3]
```diff
@@ -130,3 +130,18 @@ func (api *API) validateAdminRequest(username string) error {
 	}
 	return nil
 }
+
+func (api *API) extractPostForms(formNames []string, c *gin.Context) map[string]string {
+	var (
+		forms = make(map[string]string)
+	)
```
Comment (bobheadxi) [inline api/utils.go]: this can just be a one-line `forms := make(map[string]string)` 😋

File: api/routes_rtfs.go [hunk 4]
```diff
@@ -180,6 +176,7 @@ func (api *API) addFileLocallyAdvanced(c *gin.Context) {
 // AddFileLocally is used to add a file to our local ipfs node in a simple manner
 // this route gives the user back a content hash for their file immedaitely
 func (api *API) addFileLocally(c *gin.Context) {
+	forms := api.extractPostForms([]string{"hold_time"}, c)
```
Comment (postables) [inline api/routes_rtfs.go]: need to check length

File: api/utils.go [hunk 5]
```diff
@@ -130,3 +130,18 @@ func (api *API) validateAdminRequest(username string) error {
 	}
 	return nil
 }
+
+func (api *API) extractPostForms(formNames []string, c *gin.Context) map[string]string {
```
Comment (bobheadxi) [inline api/utils.go]: idea: use spread operator:

```suggestion
func (api *API) extractPostForms(c *gin.Context, formNames ...string) map[string]string {
```

**Pull request merged**